cg/meta/demultiplex/status_db_storage_functions.py

```diff
@@ -55,6 +56,19 @@ def store_flow_cell_data_in_status_db(
     store.session.commit()
 
 
+def check_if_samples_have_files(flow_cell_path: Path, sample_ids: List[str]) -> None:
+    """Check if all samples have already fastq files in the demultiplex directory.
+    Raises: MissingFilesError
+        When one of the samples has no fastq files in the flow cell
+    """
+    for sample_id in sample_ids:
+        fastq_files: Optional[List[Path]] = get_sample_fastqs_from_flow_cell(
+            flow_cell_directory=flow_cell_path, sample_internal_id=sample_id
+        )
+        if not fastq_files:
+            raise MissingFilesError(f"Sample {sample_id} has no fastq files in flow cell")
+
+
```

Comment:
Lets add some tests for `check_if_samples_have_files`

Reply:
will do as soon as we figure out its final form

Reply:
done